xdsl/tools/xdsl-opt

```diff
@@ -15,6 +17,43 @@ class OptMain(xDSLOptMain):
     def register_all_arguments(self, arg_parser: argparse.ArgumentParser):
         super().register_all_arguments(arg_parser)
 
+        #arg_parser.add_argument(
+        #    "--check-coverage",
+        #    default=False,
+        #    action='store_true',
+        #    help="Generate the xDSL code coverage for this run.")
+
+    def run(self):
+        # In order to find the `.coverage` file, as well as the `.coveragerc`,
+        # we need to step up to the `xdsl` directory. Notice that lit runs tests
+        # within the folder their are contained in.
+        while (os.getcwd().split('/')[-1] != 'xdsl'):
+            os.chdir('../')
+
+        data_file = ".coverage"
+        cov = coverage.Coverage(check_preimported=True,
+                                data_file=data_file,
+                                data_suffix='',
+                                config_file='.coveragerc',
+                                source_pkgs=['xdsl'],
+                                branch=True,
+                                auto_data=True)
+        if not os.path.exists(data_file):
+
+            # Sometimes when the `.coverage` file is not available, some tests
+            # fail with weird behavior. This ensures that the file is there.
+            cov.start()
+            cov.stop()
+            cov.save()
+
+        cov.start()
+
+        super().run()
+
+        cov.stop()
+        cov.xml_report(
+            outfile="/home/michel/opencompl/MLIR-lite/xdsl/coverage.xml")
```

Comment:
I also need to look into a way to not generate this report each time `xdsl-opt` is executed.

Reply:
Done (described above).